src/components/book-content-mixin.cjsx

```diff
@@ -29,6 +33,36 @@ module.exports =
       else
         img.onload = sizeImage
 
+  getMediaTag: (media) ->
+    # form media tag text based on tag name or data-type
+    tag = media.tagName
+    tag = media.dataset.type if media.dataset.type?
+
+    # capitalize
+    # TODO find one place for all such string formating helpers to go
+    tag = tag.charAt(0).toUpperCase() + tag.substring(1).toLowerCase()
+
+  isMediaLink: (link) ->
+    link.innerText is '[link]' and link.hash.length > 0 and link.hash.search('/') is -1
```

Comment:
Would be nice for case # 3 if we didn't open a new tab if it's already being viewed in the reference book.  I think we could do so by checking the router's context for the existence of `pageId`